test/test_integration_cluster.rb

```diff
@@ -451,6 +451,36 @@ def test_application_is_loaded_exactly_once_if_using_preload_app
     assert_equal 1, loading_app_count
   end
 
+  def test_application_is_loaded_exactly_once_if_using_fork_worker
+    cli_server "-w #{workers} --fork-worker test/rackup/write_to_stdout_on_boot.ru"
+
+    get_worker_pids 0
+    loading_app_count = @server_log.scan("Loading app").length
+    assert_equal 1, loading_app_count # loaded in worker 0
+
+    @server_log.clear
+    Process.kill :SIGURG, @pid
+
+    get_worker_pids 1, workers - 1
+    loading_app_count = @server_log.scan("Loading app").length
+    assert_equal 0, loading_app_count
+  end
+
+  def test_application_is_loaded_exactly_once_if_using_preload_app_with_fork_worker
+    cli_server "-w #{workers} --preload --fork-worker test/rackup/write_to_stdout_on_boot.ru"
+
+    get_worker_pids 0
+    loading_app_count = @server_log.scan("Loading app").length
+    assert_equal 1, loading_app_count # loaded in master
+
+    @server_log.clear
+    Process.kill :SIGURG, @pid
+
+    get_worker_pids 1, workers - 1
+    loading_app_count = @server_log.scan("Loading app").length
+    assert_equal 0, loading_app_count
+  end
```

Comment:
These pass on master as well, I added it to ensure I don't regress on this behaviour. I've also done some puts debugging to verify this manually.